lib/inets/src/http_client/httpc_handler.erl

```diff
@@ -622,12 +622,12 @@ do_handle_info({ssl_closed, _}, State = #state{request = undefined}) ->
     {stop, normal, State};
 
 %%% Error cases
-do_handle_info({tcp_closed, _}, #state{session = Session0} = State) ->
+do_handle_info({tcp_closed, _}, #state{session = Session0} = State) when Session0 =/= undefined->
```

Comment:
It is a bug if  Session0 is undefined at this point.  I think it would be less confusing to assert it by matching session0 = #session{socket = Socket} in the head than checking it is not undefined (that happens to be the record default value).